### PR TITLE
feat: return runtime versions used by the application with a doctor hook

### PR DESCRIPTION
**Pull request by zimeg**
### Summary

This PR introduces a `doctor` hook that returns runtime versions for the application's execution environment in a structured way. The `doctor_payload` demonstrate the expected response shape!

The `python` version is included in this response, as well as the `implementation` type and `compiler` version. All gathered from [the standard `platform` package](https://docs.python.org/3/library/platform.html#platform.python_version).

### Preview

doctor

### Testing

With a custom build of the CLI, setup a project to use the `doctor` hook:

```sh
$ cd python-slack-hooks
$ git checkout feat-doctor-hook
$ source .venv/bin/activate        # Activate your virtual environment
$ ./scripts/build_pypi_package.sh  # Prepare the package

$ slack create snak -e bolt        # Create a Python application
$ cd snak
$ python3 -m venv .venv
$ source .venv/bin/activate
$ pip install -r requirements.txt
$ pip install ~/path/to/python-slack-hooks/dist/slack_cli_hooks-0.0.0.dev2-py3-none-any.whl --force-reinstall
$ slack doctor
```

### Special notes

- IMO `implementation` and `compiler` are nice to have as it makes this section feel complete! But I imagine these are also details that can be useful in troubleshooting. Open to changing any of the included information though!

### Requirements

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/python-slack-hooks/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_and_run_tests.sh` after making the changes.

### CODE REVIEW

File: slack_cli_hooks/hooks/doctor.py
Comment (WilliamBergamin) [inline slack_cli_hooks/hooks/doctor.py:17]: Could we use the [python_version()](https://docs.python.org/3.9/library/platform.html#platform.python_version) function, it seems simpler
```suggestion
            "current": platform.python_version(),
```

File: slack_cli_hooks/hooks/doctor.py
Comment (WilliamBergamin) [inline slack_cli_hooks/hooks/doctor.py:25]: The [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler) is said to identify the compiler used for compiling Python. I don't think this will be very informative, unless a developer is compiling their own version of python (if they are, not sure how we could help them)

Providing the system information with [platform.system()](https://docs.python.org/3.9/library/platform.html#platform.system) may be more informative for maintainers try to and help a developer, let me know what you think

**Review by WilliamBergamin — CHANGES_REQUESTED** [review]
This is awesome 💯  I like the simplicity 
left some comments let me know what you think

File: slack_cli_hooks/hooks/doctor.py
Comment (zimeg) [reply slack_cli_hooks/hooks/doctor.py:17]: The `python_build()[0]` also includes the `python_revision()` which might be useful? I think it looks cool TBH but open to swapping. As is, here's how this might change:

```diff
-        python: v3.12.0:0fb18b02c8
+        python: 3.12.0
         implementation: CPython
         compiler: Clang 13.0.0 (clang-1300.0.29.30)
```

We could also include a prefixed `v` with `python_branch()` but totally your call!

File: slack_cli_hooks/hooks/doctor.py
Comment (zimeg) [reply slack_cli_hooks/hooks/doctor.py:25]: Open to removing this if you don't think it'd be useful. I was honestly looking to make a group of three and the compiler seemed interesting 👀

The system OS and architecture is also included a bit early in the `doctor` output and I think would be the same for the `platform.system()`? I really hope at least 😳 So this base might be covered already 🙌

File: slack_cli_hooks/hooks/doctor.py
Comment (WilliamBergamin) [reply slack_cli_hooks/hooks/doctor.py:17]: I think the output of `python_version()` is less ambiguous

Lets say I'm a new dev and I install python3 and then try to use the CLI but for some reason my default python path maps to python2, I think an output of `python: 2.7.1` will be easier to understand then `python: v3.7.1:0fb18b02c8`

File: slack_cli_hooks/hooks/doctor.py
Comment (WilliamBergamin) [reply slack_cli_hooks/hooks/doctor.py:25]: > The system OS and architecture is also included a bit early in the doctor output and I think would be the same for the platform.system()? I really hope at least 😳 So this base might be covered already 🙌

Just saw it now 👍 alternatively `platform.platform()` yields a string similar to this on my machine `macOS-14.4-arm64-arm-64bit` if this information if not outputted by the CLI already I think it would be more useful then [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler)

But if it is [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler) can fill the 3rd item

File: slack_cli_hooks/hooks/doctor.py
Comment (zimeg) [reply slack_cli_hooks/hooks/doctor.py:17]: Agree that `2.7.1` easier to understand but I'll poke back and say the longer version is less ambiguous 😜  For debugging having more information might be helpful, but I don't think we'd need to dive into the specific build version. Or at least I hope not...

I'll make this update but if you want to revert back to build information let me know!

File: slack_cli_hooks/hooks/doctor.py
Comment (zimeg) [reply slack_cli_hooks/hooks/doctor.py:25]: I think the GOOS / GOARCH covers the platform well enough for debugging purposes! Going to keep the compiler here for the time being

**Comment by zimeg** [comment]
@WilliamBergamin appreciate both you and your reviewing. Thank you! I'm a bit excited for this hook 🪝😄

**Pull request merged**
